engines/private/private.cpp

```diff
@@ -0,0 +1,1051 @@
+                else if (selectDossierPrevSuspect(mousePos))
+                    break;
+
+                selectPauseMovie(mousePos);
+                selectPhoneArea(mousePos);
+                selectPoliceRadioArea(mousePos);
+                selectAMRadioArea(mousePos);
+                selectLoadGame(mousePos);
+                selectSaveGame(mousePos);
+                if (!_nextSetting)
+                    selectMask(mousePos);
+                if (!_nextSetting)
+                    selectExit(mousePos);
+                break;
+
+            case Common::EVENT_MOUSEMOVE:
+                changeCursor("default");
+                if      (cursorPauseMovie(mousePos)) {}
+                else if (cursorMask(mousePos))       {}
+                else if (cursorExit(mousePos))       {}
+                //
+                break;
+
+            default:
+            break;
+            }
+        }
+
+        checkPoliceBust();
+
+        // Movies
+        if (_nextMovie != NULL) {
+            removeTimer();
+            _videoDecoder = new Video::SmackerDecoder();
+            playVideo(*_nextMovie);
+            _nextMovie = NULL;
+            continue;
+        }
+
+        if (_nextVS != NULL && *_currentSetting == kMainDesktop) {
+            loadImage(*_nextVS, 160, 120);
+        }
+
+        if (_videoDecoder) {
+            if (_videoDecoder->getCurFrame() == 0)
+                stopSound(true);
+            if (_videoDecoder->endOfVideo()) {
+                _videoDecoder->close();
+                delete _videoDecoder;
+                _videoDecoder = nullptr;
+            } else if (_videoDecoder->needsUpdate()) {
+                drawScreen();
+            }
+            continue;
+        }
+
+        if (_nextSetting != NULL) {
+            removeTimer();
+            debugC(1, kPrivateDebugFunction, "Executing %s", _nextSetting->c_str());
+            clearAreas();
+            _currentSetting = _nextSetting;
+            loadSetting(_nextSetting);
+            _nextSetting = NULL;
+            execute(prog);
+            changeCursor("default");
+            drawScreen();
+        }
+
+        g_system->updateScreen();
+        g_system->delayMillis(10);
+    }
+    return Common::kNoError;
+}
+
+void PrivateEngine::clearAreas() {
+    _exits.clear();
+    _masks.clear();
+    _loadGameMask = NULL;
+    _saveGameMask = NULL;
+    _policeRadioArea = NULL;
+    _AMRadioArea = NULL;
+    _phoneArea = NULL;
+    _dossierNextSuspectMask = NULL;
+    _dossierPrevSuspectMask = NULL;
+}
+
+void PrivateEngine::startPoliceBust() {
+    // This logic was extracted from the binary
+    int policeIndex = variables.getVal(kPoliceIndex)->u.val;
+    int r = _rnd->getRandomNumber(0xc);
+    if (policeIndex > 0x14) {
+        policeIndex = 0x15;
+    }
+    _maxNumberClicks = r + 0x10 + (policeIndex * 0xe) / -0x15;
+    _sirenWarning = _rnd->getRandomNumber(0x7) + 3;
+    _numberClicks = 0;
+    if(_sirenWarning >= _maxNumberClicks)
```

Comment:
formatting.

This is where I stopped my 3rd review.

Reply:
Fixed.